src/Mono.Android/Android.Runtime/MamXmlParser.cs

```diff
@@ -0,0 +1,159 @@
+// File must be "stand-alone"; it's included by
+// `tools/remap-mam-json-to-xml`
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Xml;
+
+using ReplacementTypesDict              = System.Collections.Generic.Dictionary<string, string>;
+using ReplacementMethodsDictStrings     = System.Collections.Generic.Dictionary<string, string>;
+using ReplacementMethodsDictStructured  = System.Collections.Generic.Dictionary<
+	(string SourceType, string SourceName, string? SourceSignature),
+	(string? TargetType, string? TargetName, string? TargetSignature, int? ParamCount, bool IsStatic)
+>;
+
+namespace Android.Runtime {
+
+	class MamXmlParser {
```

Comment:
If we went with the C# code generation idea, maybe this class could write a `.cs` file instead of XML? We don't have any Roslyn source-generators yet, so that would probably be the simplest route.